### DISCUSSION TITLE
Unable to set cookies domain

### DISCUSSION THREAD

**Discussion by Sandeep06Dev** [vercel/next.js#21965] [Help]
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

12.16.2

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**How are you deploying your application?**

next start

**Describe the Bug**

I have follow similar pattern like https://nextjs.org/docs/api-routes/api-middlewares everything working fine BUT unable to set domain of that cookie even I am able to set path or httpOnly for not the domain .

The created cookies always show subdomain.example.com as domain in chrome/devtools but I want to set the domain for example.com
 

**Expected Behavior**

Should be domain set to examples.com instead of subdomin.example.com as i am setting cookies with domain=.example.com;

**To Reproduce**

 follow similar pattern like https://nextjs.org/docs/api-routes/api-middlewares
and add domain=.example.com (or any parent domain)  while setting cookies.

Note: The nextjs project is hosted on subdomain e.g. subdomain.example.com
                                                                            The main domain is example.com 
                                                                            And i want to set cookies for example.com both domain origin are same example.com.

**Comment by EhteshamAnwar** [comment]
@Sandeep0695  found any solution to this ?
